service/worker/deployment/deployment_workflow.go

```diff
@@ -175,6 +176,11 @@ func (d *DeploymentWorkflowRunner) handleRegisterWorker(ctx workflow.Context, ar
 		d.lock.Unlock()
 	}()
 
+	// check if we can add more task-queues in the deployment
+	if len(d.DeploymentLocalState.TaskQueueFamilies)+1 == int(d.DeploymentLocalState.MaxTaskQueues) {
```

Comment:
If we add this check to the validation step, it would presumable eliminate the need of making the limit part of the wf state (because validate func is not added to wf history and changing the limit will not break wf determinism), right?

Reply:
I think you are right here + I like the idea of having the limit being sent as an arg in the update itself - wf determinism slipped out of my mind and it's possible since the dynamic config can be altered by anyone even when the wf is going on